hibernate-core/src/main/java/org/hibernate/collection/spi/AbstractPersistentCollection.java

```diff
@@ -652,6 +656,12 @@ private void throwLazyInitializationException(String message) {
 		);
 	}
 
+	public static void checkPersister(boolean collectionInitialized, CollectionPersister persister) {
+		if ( !collectionInitialized && persister == null ) {
+			throw new HibernateException( "Attempt to access a lazy collection from within an illegal context (e.g. a lifecycle listener method)" );
```

Comment:
I believe the error should be the following:

```suggestion
			throwLazyInitializationException( "collection is being removed" );
```